pkg/machine/convert.go

```diff
@@ -278,3 +284,17 @@ func GetAPIV2NodeCloudSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv1.Node
 
 	return cloudSpec, nil
 }
+
+func extractSpotInstanceConfigs(config *aws.RawConfig) (*string, *string, *bool) {
+	if config.IsSpotInstance != nil &&
+		*config.IsSpotInstance &&
+		config.SpotInstanceConfig != nil {
+		if config.SpotInstanceConfig != nil {
```

Comment:
Nit: you already check `config.IsSpotInstance != nil` above.

Reply:
😂😬